pkg/framework/spi/kms/api.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package kms
+
+import "github.com/google/tink/go/keyset"
+
+// Provider interface for Crypto ctx
+type Provider interface {
+}
+
+// Creator method to create new pluggable KMS service
+type Creator func(prov Provider) (KMS, error)
+
+// KMS is an interface defining a pluggable KMS operations provider to Aries agents.
+// It's responsible for managing keys and provide KMS specific crypto operations.
+type KMS interface {
+	// Encrypt will encrypt msg and aad using a matching AEAD primitive in kh
+	// returns:
+	// 		cipherText in []byte
+	//		nonce in []byte
+	//		error in case of errors during encryption
+	// TODO  to be defined according to existing cloud KMSs and local mobile keystores..
+	Encrypt(msg, aad []byte, keyHandle *keyset.Handle) ([]byte, []byte, error)
```

Comment:
The KMS Encrypt/Decrypt functions here has [same](https://github.com/hyperledger/aries-framework-go/pull/958/files#diff-6d57c757cc93b93f22de0729f28d5a52R27) definition as Crypto interface. Why we have same functionality with 2 interfaces ? Is KMS subset of Crypto ?

Reply:
I agree - if Crypto is a subset of KMS, then KMS *is a* Crypto. The method definitions shouldn't be duplicated.

Reply:
I don't want Crypto to be defined as a subset of KMS or vice versa. They should be treated as two different services providing their own operations. weather KMS offers some crypto operations, this is up to the KMS implementation, for example current KMSs in Tink don't offer Sign/Verify functions but do offer Encrypt/Decrypt. 

So for both interfaces, although they may have the same functions, should define them separately.

Reply:
Well, your model strongly suggests there is directional IS-A relationship between KMS and Crypto. If not all implementations of KMS will offer Crypto operations then those should be removed from the KMS.